File: serenity-app/src/main/java/us/nineworlds/serenity/injection/InjectingJob.java
package us.nineworlds.serenity.injection;

import com.birbit.android.jobqueue.Job;
import com.birbit.android.jobqueue.Params;
import us.nineworlds.serenity.common.injection.SerenityObjectGraph;

public abstract class InjectingJob extends Job {

  public InjectingJob() {
    super(new Params(500).requireNetwork());
    SerenityObjectGraph.Companion.getInstance().inject(this);
  }
}

File: serenity-android-common/src/main/kotlin/us/nineworlds/serenity/common/android/injection/InjectingJob.kt
package us.nineworlds.serenity.common.android.injection

import com.birbit.android.jobqueue.Job
import com.birbit.android.jobqueue.Params
import toothpick.Toothpick
import us.nineworlds.serenity.common.annotations.InjectionConstants

@Suppress("LeakingThis")
abstract class InjectingJob : Job(Params(500).requireNetwork()) {
  init {
    val jobScope = Toothpick.openScope(InjectionConstants.APPLICATION_SCOPE)
    Toothpick.inject(this, jobScope)
  }

}
